src/Imagick/Image.php

```diff
@@ -304,7 +304,11 @@ public function rotate($angle, ColorInterface $background = null)
         try {
             $pixel = $this->getColor($background);
 
-            $this->imagick->rotateimage($pixel, $angle);
+            $imageWidth = $this->imagick->getImageWidth();
+            $imageHeight = $this->imagick->getImageHeight();
+
+            $this->imagick->rotateImage($pixel, $angle);
+            $this->imagick->setImagePage($imageWidth, $imageHeight, 0, 0);
```

Comment:
Setting the image page to the dimensions before the resize seems to be wrong because the image size grows on rotation.

From the [ImageMagick documentation](https://imagemagick.org/script/command-line-options.php#repage): 
> A given a canvas size of zero such as '0x0' forces it to recalculate the canvas size so the image

So `->setImagePage(0, 0, 0, 0);` should do the job here.